### PR TITLE
catalyst flow UI

### PR DESCRIPTION
**Pull request by ashisherc**

### CODE REVIEW

File: app/containers/wallet/VotingPage.js [hunk 1]
```diff
@@ -38,190 +20,36 @@ type Props = {|
...
-          onCancel={this.cancel}
-          onSubmit={({ password }) =>
-            this.generated.actions.ada.votingTransaction.signTransaction.trigger({
-              password,
-              publicDeriver: selectedWallet,
-            })
-          }
-          classicTheme={this.generated.stores.profile.isClassicTheme}
-          error={this.generated.stores.wallets.sendMoneyRequest.error}
+    const { uiDialogs } = this.generated.stores;
```
Comment (SebastienGllmt) [inline app/containers/wallet/VotingPage.js]: Remember you need to disable this flow for hardware wallets

Comment (SebastienGllmt) [reply app/containers/wallet/VotingPage.js]: Also you need to disable this for Byron wallets (AKA wallets with no signing key)

File: app/components/wallet/voting/ConfirmPinDialog.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          {intl.formatMessage(messages.line1)}
+        </div>
+        <div className={classnames([styles.pinInputContainer])}>
+          <PinInput
+            setForm={form => this.setPinForm(form)}
+            disabled={false}
+            classicTheme={classicTheme}
+            pinMatches={pinValidation}
+            fieldName="pin"
+            validCheck={_pin => true}
```
Comment (SebastienGllmt) [inline app/components/wallet/voting/ConfirmPinDialog.js:88]: Looks like this isn't actually used?

File: app/components/wallet/voting/ConfirmPinDialog.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+          />
+        </div>
+      </Dialog>);
+  }
+
+  _submitForm: void => Promise<void> = async () => {
+    if (this.pinForm !== undefined) {
+      this.pinForm.submit({
+        onSuccess: async () => {
+          this.props.submit();
```
Comment (SebastienGllmt) [inline app/components/wallet/voting/ConfirmPinDialog.js]: await

File: app/components/wallet/voting/RegisterDialog.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        </div>
+      </Dialog>);
+  }
+
+  _submitForm: void => Promise<void> = async () => {
+    if (this.spendingPasswordForm !== undefined) {
+      this.spendingPasswordForm.submit({
+        onSuccess: async form => {
+          const { walletPassword } = form.values();
+          this.props.submit(walletPassword);
```
Comment (SebastienGllmt) [inline app/components/wallet/voting/RegisterDialog.js]: await

File: app/containers/wallet/dialogs/voting/RegisterDialogContainer.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    }
+    return (
+      <RegisterDialog
+        progressInfo={votingStore.progressInfo}
+        submit={async (walletPassword: string) => {
+          try {
+            await this.generated.actions.ada.votingTransaction.createTransaction.trigger(
+              walletPassword
+              );
+            submit();
```
Comment (SebastienGllmt) [inline app/containers/wallet/dialogs/voting/RegisterDialogContainer.js]: await

File: app/containers/wallet/dialogs/voting/TransactionDialogContainer.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+          }
+          getTokenInfo={genLookupOrFail(this.generated.stores.tokenInfoStore.tokenInfo)}
+          onCancel={cancel}
+          goBack={goBack}
+          onSubmit={async ({ password }) => {
+              await this.generated.actions.ada.votingTransaction.signTransaction.trigger({
+                password,
+                publicDeriver: selectedWallet,
+              })
+              submit();
```
Comment (SebastienGllmt) [inline app/containers/wallet/dialogs/voting/TransactionDialogContainer.js]: await

File: app/components/wallet/voting/ConfirmPinDialog.js (hunk 2 above)
Comment (ashisherc) [reply app/components/wallet/voting/ConfirmPinDialog.js:88]: nope, but I have kept this in the PinInput component so any future PinInput related UI may use it

File: app/components/wallet/voting/QrCodeDialog.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        </div>
+
+        <div className={styles.lineBold}>
+          {intl.formatMessage(messages.line3)}
+        </div>
+
+        <div className={classnames([styles.qrCodeContainer, styles.lastItem])}>
+          {votingKey !== null ? (
+            <div className={styles.qrCode}>
+              <QRCode
```
Comment (SebastienGllmt) [inline app/components/wallet/voting/QrCodeDialog.js]: Use `QrCodeWrapper` instead (this was added after you started this PR)

**Review by SebastienGllmt — APPROVED** [review]

**Pull request merged**